src/api/user.controller.js

```diff
@@ -0,0 +1,78 @@
+const { userService } = require('../services/user.service.js');
+
+const getUsers = async (req, res) => {
+  const users = await userService.getUsers();
+
+  res.json(users);
+};
+
+const createUser = async (req, res) => {
+  const name = req.body.name;
+
+  if (!name || typeof name !== 'string' || name.trim() === '') {
+    return res
+      .status(400)
+      .json({ message: 'Name is required and must be a valid string.' });
+  }
+
+  const user = await userService.createUser(name.trim());
+
+  res.status(201).json(user);
+};
+
+const getUserById = async (req, res) => {
+  const user = await userService.getUserById(+req.params.id);
+
+  if (!user) {
+    return res.status(404).json({ message: 'User not found.' });
+  }
+
+  res.json(user);
+};
+
+const updateUser = async (req, res) => {
+  const { name } = req.body;
```

Comment:
Similar to the `createUser` function, ensure that all necessary fields in the `req.body` are validated in the `updateUser` function. This will help maintain data integrity.